Veneer/Classes/UIBezierPath+Outline.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  UIBezierPath+Outline.swift
+//  Pods
+//
+//  Created by Sam Watts on 10/02/2017.
+//
+//
+
+import UIKit
+
+extension UIBezierPath {
+    
+    //code adapted from http://stackoverflow.com/a/35711310/1138900
+    //works through top points from left to right, then bottom points right to left, adding intersection points if crossing the axis of previous point
+    
+    convenience init(outliningViewFrames frames: [CGRect]) {
+
+        self.init()
+        
+        //nothing to do for empty frames
+        guard frames.isEmpty == false else { return }
+        
+        //check for simple case of a single frame
+        guard frames.count != 1 else { self.append(UIBezierPath(rect: frames[0])); return }
+        
+        // top left and right corners of each view
+        // sorted from left to right, top to bottom
+        var topPoints:[CGPoint] = frames.reduce([]) { $0 + [CGPoint(x: $1.minX, y: $1.minY), CGPoint(x: $1.maxX, y: $1.minY)] }
+        topPoints = topPoints.sorted(by: { $0.x == $1.x ? $0.y < $1.y : $0.x < $1.x })
+        
+        // trace top line from left to right
+        // moving up or down when appropriate
+        var previousPoint = topPoints.removeFirst()
+        self.move(to: previousPoint)
+        for point in topPoints {
+            
+            //guard against non outline points (e.g. corner points withing other rects)
+            guard point.y == previousPoint.y
+                || point.y < previousPoint.y
```

Comment:
good question, from what I can tell they are the same precedence so would be lazily matched depending on current state. I think there are really 3 separate cases here so have added parenthesis to make it clearer